site/topics/logic/combinational-logic.rst

```diff
@@ -10,6 +10,146 @@ Combinational Logic
 Decoders
 ========
 
+* Consider a situation where an input signal controls two outputs, where only one should be active at a time
+
+    * When the input signal is low, one of the two outputs is high
+    * When the input signal is high, the other output should be high
+
+
+.. list-table:: Single Bit Decoder
+    :widths: auto
+    :align: center
+    :header-rows: 1
+
+    * - Input
+      -
+      - Output a
+      - Output b
+    * - ``0``
+      -
+      - ``1``
+      - ``0``
+    * - ``1``
+      -
+      - ``0``
+      - ``1``
+
+
+.. note::
+
+    Until now, all truth tables shown included boolean operators and the tables were completed to describe how the
+    operators act.
+
+    The above truth table does not include any specific boolean operators. Here, the truth table is describing the
+    desired functionally without specifying any operators.
+
+
+* Based on the truth table, how could this functionality be achieved with boolean operators?
+
+    * Output a is the inversion of the input
+    * Output b is simply the input
+
+
+.. figure:: single_bit_decoder.png
+    :width: 500 px
+    :align: center
+
+    A single bit decoder. At any point in time, a single output is always high. The specific output that is high is
+    controlled by the input signal.
+
+
+* This is an example of a single bit decoder
+
+    * The one input signal can be thought of as a single bit
+    * The input signal is "decoded" to control the output signals
+
+
+More than One Input Bit
+-----------------------
+
+* Now consider a situation where four outputs need to be controlled
+* A single input can only have one of two states, low/high
+
+    * Thus,  more than two input signals would be needed to decode to four outputs
+
+
+* Two input signals can have a total of four unique states
+
+    * Two inputs are sufficient to decode to four output signals
+
+
+.. list-table:: Two Bit Decoder
+    :widths: auto
+    :align: center
+    :header-rows: 1
+
+    * - Input a
+      - Input b
+      -
+      - Output 0
+      - Output 1
+      - Output 2
+      - Output 3
+    * - ``0``
+      - ``0``
+      -
+      - ``1``
+      - ``0``
+      - ``0``
+      - ``0``
+    * - ``0``
+      - ``1``
+      -
+      - ``0``
+      - ``1``
+      - ``0``
+      - ``0``
+    * - ``1``
+      - ``0``
+      -
+      - ``0``
+      - ``0``
+      - ``1``
+      - ``0``
+    * - ``1``
+      - ``1``
+      -
+      - ``0``
+      - ``0``
+      - ``0``
+      - ``1``
+
+
+* The trick to creating a decoder with more than one bit is to use and gates with specific inputs inverted
+
```

Comment:
The trick? This statement seems to come out of nowhere. There's no motivation for why decoders are even relevant beyond an abstract problem statement.  
And this only makes a 1/n decoder.

Reply:
I reworked the section on more than 1 input significantly. 

I tried to solidify up the motivation, but it is still on the abstract side as it's some arbitrary 4 circuits. If the topic doing the intro to memory comes before this one, there is easier motivation --- addressing memory. 

Let me know if you think it's fine as is now, or if you think hitting memory directly would be better. I suppose the only minor fear is that the decoder could be _too_ associated with memory, even though it's a far more general tool. I am also not clear on how everything would fit together in the end if the into memory topic comes before this; this may be easier to figure out _after_ the rest of this topic and the intro memory topic is hit.